### DISCUSSION TITLE
Themes - Horizontal Layout

### DISCUSSION THREAD

**Discussion by jianmc** [oqtane/oqtane.framework#3733] [General]
I'm trying to template a <Pane> where the modules will be added in a horizontal rather than vertical fashion.

Looking at the layout, there's extra <a> links that when deleted cause the layout to render the way I would like.

So, is there a way to do this or is there a way to remove elements like: ```<a id="40"></a>```  ?

For reference, my theme would contain something like this:
```html
<div class="row">
    <div class="col-md-12">
        <div class="row">
            <Pane Name="Top 100%" />
        </div>
    </div>
</div>
``` 

And the container would be:
```html
<div class="col">
    <div class="d-flex flex-nowrap">
        <ModuleActions />
    @if (_title && ModuleState.Title != "-")
    {
        <h2><ModuleTitle /></h2>
    }
        <div class="container-fluid">
            <ModuleInstance />
        </div>
    </div>
</div>
``` 
Rendered example:
```html
<div class="row"><div class="col-md-12"><div class="row"><!--!--><!--!--><a id="40"></a><!--!-->
    <!--!--><!--!--><!--!--><div class="col"><div class="d-flex flex-nowrap"><!--!--><div class="container-fluid"><!--!--><!--!--><!--!--><!--!--><!--!-->...</div></div></div><!--!--><!--!-->
    <!--!--><!--!--><!--!--><div class="col"><div class="d-flex flex-nowrap"><!--!--><div class="container-fluid"><!--!--><!--!--><!--!--><!--!-->...</div></div></div><!--!--><a id="49"></a><!--!-->
    <!--!--><!--!--><!--!--><div class="col"><div class="d-flex flex-nowrap"><!--!--><h2><!--!--><span class="app-moduletitle"><!--!-->HtmlText</span></h2><div class="container-fluid"><!--!--><!--!--><!--!--><!--!--> </div></div></div></div></div></div>
``` 

Any advice is greatly appreciated!

**Comment by sbwalker** [comment]
The ```<a id="40"></a>``` is coming from the Oqtane.Client\UI\ContainerBuilder.razor - it is an anchor tag which allows a user to use a url fragment (ie. https://domain.com#40) to scroll to specific module on a page. Currently it is a core feature which cannot be disabled. It probably should have a unique class on the element so that it is possible to style it via CSS (ie. make it disappear).

**Comment by jianmc** [comment]
It looks like adding a "col-auto" would work.

How about adding a class to those links that can be part of the theme?

**Reply by sbwalker** [reply]
In order to ensure this introduced in a generic and backward compatible manner the anchor would use an Oqtane specific class ie. ```<a id="40" class="app-anchor"></a>```. It would have no CSS style defined by default. However third party developers could include their own style which would only affect their own applications.

**Comment by jianmc** [comment]
I agree this would work.  I don't see the need for a different style
depending on the container applied.

My objective is to give the design team the ability to place the components
in either direction without involving me for simple tests.  The more they
can be empowered to design on their own, the better.  I have no issue doing
a starter layout and a final layout, just want to remove myself from the
steps in between.
